source/opt/def_use_manager.h

```diff
@@ -92,23 +96,34 @@ class DefUseManager {
 
  private:
   // Analyzes the defs and uses in the given |module| and populates data
-  // structures in this class.
+  // structures in this class. Does nothing if |module| is nullptr.
   void AnalyzeDefUse(ir::Module* module);
 
   // Clear the internal def-use record of a defined id if the given |def_id| is
   // recorded by this manager. This method will erase both the uses of |def_id|
   // and the |def_id|-generating instruction's use information kept in this
-  // manager, but not the operands in the original instructions.
+  // manager, but not the operands in the original instructions. Does nothing
+  // if |def_id| was not recorded before.
   void ClearDef(uint32_t def_id);
 
-  using ResultIdToUsedIdsMap =
-      std::unordered_map<uint32_t, std::vector<uint32_t>>;
+  // Clear the internal def-use record of the given instruction |inst|. This
+  // method will update the use information of the operand ids of |inst|. The
+  // record: |inst| uses an |id|, will be removed from the use records of |id|.
+  // If |inst| defines an result id, the use record of this result id will also
+  // be removed. Does nothing if |inst| was not analyzed before.
+  void ClearInst(ir::Instruction* inst);
+
+  // Erases the records that a given instruction |inst| uses its operand ids.
+  void EraseUseRecordsOfOperandIds(const ir::Instruction* inst);
+
+  using InstToUsedIdsMap =
```

Comment:
Type declaration should appear at the very beginning of each section (after `private:` for this case).